### PR TITLE
Create cmake-multi-platform.yml

### PR DESCRIPTION
**Pull request by Argus74**

### CODE REVIEW

**Review by alexchuber — APPROVED** [review]
LGTM. This is gonna help our workflow tremendously. Thanks!

Once this gets merged, I can open a dummy, breaking PR to make sure everything is good. Might also be worth looking into a build check targeted at whatever MacOS Baron is using. (Unless it'd take more than 20 minutes to add to our checks... in which case, maybe not worth the hassle. Windows is good enough.)

**Comment by Argus74** [comment]
Just edited the file to make sure it only does these checks on dev instead of main. If the dev branch is good to go, then main will be good to go when we merge from dev to main.

**Review by alexchuber — APPROVED** [review]

**Review by adishchukufl — APPROVED** [review]

**Pull request merged**
